src/main/java/org/threadly/concurrent/PriorityScheduler.java

```diff
@@ -763,232 +639,543 @@ protected void addToExecuteQueue(OneTimeTaskWrapper task) {
    * 
    * @param task {@link TaskWrapper} to queue for the scheduler
    */
-  protected void addToScheduleQueue(TaskWrapper task) {
-    if (shutdownStarted.get()) {
+  protected void addToScheduleQueue(TaskPriority priority, TaskWrapper task) {
+    if (workerPool.isShutdownStarted()) {
       throw new RejectedExecutionException("Thread pool shutdown");
     }
     
-    switch (task.priority) {
-      case High:
-        highPriorityConsumer.addScheduled(task);
-        break;
-      case Low:
-        lowPriorityConsumer.addScheduled(task);
-        break;
-      default:
-        throw new UnsupportedOperationException();
-    }
+    getQueueManager(priority).addScheduled(task);
+  }
+  
+  @Override
+  protected void finalize() {
```

Comment:
Here is the finalizer implementation for PriorityScheduler and the extending class PrioritySchedulerStatisticTracker